hashicorp/consul-k8s
# CNI File watcher and pre applying rules setup

Changes proposed in this PR:
- increase limits for CNI plugin so that it runs on GKE
- add annotations for transparent proxy status (enabled, waiting)
- Initial setup (CNI_ARGS) for getting information to the CNI plugin
- file watcher for config file changes and for when the config file does not exists
- added wait for annotation to be used before applying ipconfig traffic redirection

How I've tested this PR:

- unit tests
- manual testing

How I expect reviewers to test this PR:

- You can run this in kind by disabling kinds default CNI, installing calico first and installing consul with cni turn on.

Or you can clone my [cni-demo repo](https://github.com/curtbushko/cni-demo/tree/main/consul-cni-on-kind) and run:

`cd consul-cni-on-kind; make create-cluster deploy-calico deploy-consul hashicups`

Note: the consul-cni will crashloop as I need to add the file watcher to it.

Checklist:
  > HashiCorp engineers only, community PRs should not add a changelog entry.
  > Entries should use present tense (e.g. Add support for...)
